core/src/main/java/com/graphhopper/routing/util/BikeFlagEncoder.java

```diff
@@ -27,7 +27,7 @@
  */
 public class BikeFlagEncoder extends BikeCommonFlagEncoder {
     public BikeFlagEncoder() {
-        this(4, 2, 0);
```

Comment:
So previously we were using different defaults (e.g. `4,2,0` for bike and `5,5,0` for car). Why is this not necessary anymore? I am just curious.

Reply:
The defaults are the same but it is now no longer necessary to repeat them in two constructors and only have them in the PMap constructor.

Reply:
Ah I see :+1:

Reply:
Should we move this code that is currently in the `PMap` constructor 

```java
        blockBarriersByDefault(properties.getBool("block_barriers", false));
        blockPrivate(properties.getBool("block_private", true));
        blockFords(properties.getBool("block_fords", false));
```

also into the `(int, double, int)` constructor? Otherwise there might be a difference between the two? And especially the default constructor was is using the `PMap` constructor (including these calls) while it was using the other one before.

Reply:
Or actually it might be easier to do this correctly this way:

```java
blockBarriersByDefault(properties.getBool("block_barriers", isBlockBarriersByDefault()));
```

instead of 

```java
blockBarriersByDefault(properties.getBool("block_barriers", false))
```

This way the `PMap` constructor uses the same default as the `int, double, int` constructor and really only changes the default if the `PMap` has the corresponding entry.

Reply:
Sorry this is a bit off-topic here, but looking at these constructors has confused be quite a few times :)

Reply:
Yeah, these flag encoders "evolved" a lot and I'm unsure how much energy we should put into refactoring when the aim for 2.0 is to remove them

> also into the (int, double, int) constructor?

But we won't have a PMap there where it would make sense or what do you mean?

> Sorry this is a bit off-topic here

No worries I should have done these things on master instead :)

Reply:
> Yeah, these flag encoders "evolved" a lot and I'm unsure how much energy we should put into refactoring when the aim for 2.0 is to remove them

Ok yes, but we still need to make sure the code is correct/unchanged and right now I think this is hard to verify: For example `CarFlagEncoder()` previously was invoking the `(int, double, int)` constructor, but now its invoking the `CarFlagEncoder(PMap)` constructor which compared to the other constructor has an additional call to `blockFords(properties.getBool("block_fords", false)`. To find out whether or not that makes a difference we have to look at the default value in `AbstractFlagEncoder` and as it seems the default value for `blockFords` is `true` (not `false`). Or to put it shortly: On master `new CarFlagEncoder().isBlockFords()` is `true` and on this branch its `false`. I am not sure if this was intended, but for me this is reason enough to clean this up, because it already took too long to find out what the code does and just looking at the code tells me its suspicious because the default values are scattered across multiple locations and the constructors are not chained properly. And sure flag encoders shall be removed at some point, but it will be even harder to remove or refactor them when they are already hard to understand so I would count such clean up as a first step of removing them :)

Maybe we can remove the `int,double,int` constructor from `AbstractFlagEncoder` and only use `AbstractFlagEncoder(PMap)` instead and do the `blockFords(properties.getBool...` stuff there?

Reply:
Ok, will see if I can remove these (hopefully unrelated) changes here.